query_optimizer/ExecutionGenerator.cpp

```diff
@@ -1575,21 +1579,12 @@ void ExecutionGenerator::convertSort(const P::SortPtr &physical_sort) {
   merged_runs_destination_proto->set_relational_op_index(merge_run_operator_index);
   sorted_output_destination_proto->set_relational_op_index(merge_run_operator_index);
 
-  // Do not add merged_runs_relation into 'temporary_relation_info_vec_'
-  // and create the DropTableOperator for it at the end. Instead, add the drop
-  // operator right here, because the relation won't be used by any other operator.
-  const QueryPlan::DAGNodeIndex drop_merged_runs_index =
-      execution_plan_->addRelationalOperator(
-          new DropTableOperator(
-              query_handle_->query_id(),
-              *merged_runs_relation,
-              optimizer_context_->catalog_database(),
-              false /* only_drop_blocks */));
-  execution_plan_->addDirectDependency(
-      drop_merged_runs_index,
-      merge_run_operator_index,
-      true /* is_pipeline_breaker */);
-
+  // REVIEW(Shixuan): There might be an issue with the original code: if a throw
+  // happens before the plan is executed, merged_runs_relation will not be dropped
+  // by dropAllTemporaryRelations() if not added to temporary_relation_info_vec,
+  // which might cause "RelationNameCollision".
+  temporary_relation_info_vec_.emplace_back(merge_run_operator_index,
```

Comment:
@shixuan-fan I prefer the old way that deletes the tmp sorted relation eagerly, as the resultant blocks belonged to that relation would be huge.

No need to worry about the relation name collision, as it is created per query basis.